josh-proxy/src/bin/josh-proxy.rs

```diff
@@ -615,6 +627,15 @@ async fn call_service(
         }
     }
 
+    let span = tracing::span!(tracing::Level::TRACE, "hyper_cgi");
+    let _enter = span.enter();
+    let mut context_propagator = HashMap::<String, String>::default();
+    let context = span.context();
+    global::get_text_map_propagator(|propagator| {
+        propagator.inject_context(&context, &mut context_propagator);
+    });
```

Comment:
can this be wrapped into something with a meaningful name? like `propagate_span` or something

Reply:
I am not sure it makes sense. From my understanding, span and _enter need to be declared in this context to link stuff properly. Maybe the three remaining lines then?